creator-node/src/AsyncProcessingQueue.js

```diff
@@ -0,0 +1,185 @@
+const Bull = require('bull')
+const { logger: genericLogger } = require('./logging')
+const config = require('./config')
+const redisClient = require('./redis')
+
+// Processing fns
+const {
+  handleTrackContentRoute: trackContentUpload
+} = require('./components/tracks/tracksComponentService')
+
+const MAX_CONCURRENCY = 100
+const EXPIRATION_SECONDS = 86400 // 24 hours in seconds
+const PROCESS_NAMES = Object.freeze({
+  trackContentUpload: 'trackContentUpload'
+})
+const PROCESS_STATES = Object.freeze({
+  IN_PROGRESS: 'IN_PROGRESS',
+  DONE: 'DONE',
+  FAILED: 'FAILED'
+})
+
+class AsyncProcessingQueue {
```

Comment:
could you add some comments around the intent of this queue. it's very similar to the old FileProcessingQueue but it's intended to be different